### PR TITLE
chore(ci): split page-screenshots into build + publish workflows 

### PR DESCRIPTION
**Pull request by compojoom**
> Split the screenshot job,
> trim its permissions,
> sandbox the install.

## What it solves

The page-screenshots workflow combined an after-deploy trigger with a head-ref checkout, broad write permissions, and execution of repo-resident scripts and local composite actions from the checked-out workspace. That meant any change to a script or composite action under `.github/` would run with the workflow's full token on every PR, including from branches that hadn't been reviewed yet.

https://linear.app/safe-global/issue/WA-2338

Splits the workflow into two halves linked by `workflow_run`:

**`page-screenshots-build.yml`** runs after `Web Deploy to dev/staging` completes (so the preview URL is live). It has `permissions: {}` at workflow level, no `secrets.*` references, and no cache-write steps. Playwright is installed into `$RUNNER_TEMP/playwright-install` (outside the checked-out workspace) with `--ignore-scripts` and pinned to `playwright@1.60.0`, so neither the repo's own `package.json` nor any lifecycle script can influence what gets installed. The job uploads a single PNG artifact and exits.

**`page-screenshots-publish.yml`** runs on `workflow_run` from the build, checks out only the default branch (`dev`), and consumes the named artifact via the GitHub API. The artifact is resolved by ID bound to the originating `workflow_run.id` so a stale or unrelated artifact cannot be picked up. PNG filenames are validated against `^[a-z0-9_]+__(desktop|mobile)\.png$`, and total uncompressed size is capped at 50 MB. The PR number is resolved server-side via `listPullRequestsAssociatedWithCommit`, the storage-branch name is derived from that PR number (`pr-<n>-screenshots`), and the existing `upload-to-storage-branch` composite action is invoked from the `dev` checkout. A sticky PR comment is posted with a stale-SHA banner if the PR HEAD moved between build and publish.

Other changes:
- `capture-page-screenshots.js` writes filenames as `<routeSlug>__desktop.png` to match the publish-side allowlist.
- `upload-to-storage-branch` adds `set -euo pipefail` and constrains `storage_branch` to a known shape so callers cannot pass arbitrary refs.

## How to test it

`workflow_run` listeners only fire from the workflow files as they exist on the default branch, so this PR cannot exercise the new chain on itself. Two options:

1. **Recommended — open a follow-up PR after merge.** Once this lands on `dev`, open any small PR that changes a file under `apps/web/src` or `packages/`. Expected: `Web Deploy to dev/staging` runs → `Page Screenshots Build` runs and uploads the `page-screenshots` artifact → `Page Screenshots Publish` resolves the artifact, pushes to `pr-<n>-screenshots`, and posts a sticky comment with the screenshots. Verify in the publish run logs that the resolved artifact ID is bound to the originating run and every PNG passes the filename allowlist.
2. **Pre-merge dry run.** Push these workflow files onto a throwaway long-lived branch (e.g. `screenshots-test`), point that branch at `head_branch` filters as needed, and open a PR targeting it. Heavier — only worth it if option 1's feedback loop is too slow.

Edge cases worth covering in the follow-up PR run:
- No relevant source files changed → build skips the upload step → publish deletes any prior screenshot comment.
- Force-push between build and publish → publish prefixes the comment with `**STALE**`.
- A second PR with the same head SHA → publish fails closed (ambiguous PR resolution).

## Visual summary

```mermaid
flowchart LR
  subgraph Build [page-screenshots-build.yml]
    direction TB
    B1[workflow_run: Web Deploy completed]
    B2[permissions: empty]
    B3[checkout head ref]
    B4[install Playwright in fresh dir<br/>--ignore-scripts, pinned]
    B5[run screenshot scripts]
    B6[upload-artifact: PNGs only]
    B1 --> B2 --> B3 --> B4 --> B5 --> B6
  end
  subgraph Publish [page-screenshots-publish.yml]
    direction TB
    P1[workflow_run: build completed]
    P2[checkout dev only]
    P3[resolve artifact by ID<br/>workflow_run.id binding]
    P4[validate filenames + size cap]
    P5[resolve PR via API]
    P6[upload-to-storage-branch<br/>pr-N-screenshots]
    P7[sticky PR comment]
    P1 --> P2 --> P3 --> P4 --> P5 --> P6 --> P7
  end
  Build -.workflow_run.-> Publish
```

## Checklist

- [ ] I've tested the branch on mobile
- [ ] I've documented how it affects the analytics (if at all)
- [ ] I've written a unit/e2e test for it (if applicable)

### CODE REVIEW

**Comment by PooyaRaki** [comment]
@claude review

**Review by katspaugh — DISMISSED** [review]
Thanks! 🙏

**Comment by PooyaRaki** [comment]
@claude review

File: .github/scripts/capture-page-screenshots.js [hunk 1]
```diff
@@ -6,7 +6,16 @@
 
 const fs = require('fs')
 const path = require('path')
-const { chromium } = require('playwright')
+
+// Playwright is installed into an isolated directory outside the workspace by
+// the build workflow. Resolve by absolute path so a stray local `node_modules`
+// can never shadow the pinned install.
+const playwrightPath = process.env.PLAYWRIGHT_PATH
```
Comment (PooyaRaki) [inline .github/scripts/capture-page-screenshots.js:13]: We should validate this as well.

File: .github/scripts/publish/resolve-pr.js [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+ *   skip   — "true" if there is no actionable PR (zero matches → designed skip)
+ *   number — PR number (only when skip === "false")
+ *   stale  — "true" if the PR's current HEAD differs from HEAD_SHA
+ *
+ * Env:
+ *   HEAD_SHA — workflow_run.head_sha
+ */
+
+module.exports = async ({ github, context, core }) => {
+  const headSha = process.env.HEAD_SHA
```
Comment (PooyaRaki) [inline .github/scripts/publish/resolve-pr.js:19]: nit: would be nice to validate this as well.

Comment (compojoom) [reply .github/scripts/publish/resolve-pr.js:19]: The value comes from github.event.workflow_run.head_sha, set by GitHub's event payload (not user input), and the job already gates on head_repository.full_name == github.repository. A shape check here would be cosmetic

File: .github/scripts/capture-page-screenshots.js (hunk 1 above)
Comment (compojoom) [reply .github/scripts/capture-page-screenshots.js:13]: This script is loaded from the PR's head ref by the build job, so anything we write here (including validation) can be edited on the same branch — it can't enforce a contract. PLAYWRIGHT_PATH itself is set by inline YAML in the build workflow, which workflow_run resolves from the default branch

**Review by PooyaRaki — APPROVED** [review]

**Review by agusgonzaleznic — APPROVED** [review]
